File: solution/java/day15/src/test/java/DocumentTests.java
import document.DocumentTemplateType;
import document.RecordType;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

class DocumentTests {
    @Test
    void combinationTests() {
        verifyAllCombinations(
                DocumentTemplateType::fromDocumentTypeAndRecordType,
                Arrays.stream(DocumentTemplateType.values()).map(Enum::name).toArray(String[]::new),
                Arrays.stream(RecordType.values()).map(Enum::name).toArray(String[]::new)
        );
    }
}

File: solution/kotlin/day15/src/test/kotlin/DocumentTests.kt
import document.DocumentTemplateType
import document.RecordType
import org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations
import org.junit.jupiter.api.Test

class DocumentTests {
    @Test
    fun combinationTests() {
        verifyAllCombinations(
            { document, record -> DocumentTemplateType.fromDocumentTypeAndRecordType(document, record) },
            DocumentTemplateType.entries.map { e -> e.name }.toTypedArray(),
            RecordType.entries.map { e -> e.name }.toTypedArray()
        )
    }
}